### PR TITLE
Optimize UI text measurement

### PR DESCRIPTION
**Pull request by UkoeHB**
# Objective

- Avoid cloning the `CosmicBuffer` every time you create a new text measurement.

## Solution

- Inject a buffer query when calculating layout so existing buffers can be reused.

## Testing

- I tested the `text`, `text_debug`, and `text_wrap_debug` examples.
- I did not do a performance test.

### CODE REVIEW

File: crates/bevy_ui/src/widget/text.rs [hunk 1]
```diff
@@ -47,12 +47,20 @@ pub struct TextMeasure {
     pub info: TextMeasureInfo,
 }
 
+impl TextMeasure {
+    /// Checks if the cosmic text buffer is needed for measuring the text.
+    pub fn needs_buffer(height: Option<f32>, available_width: AvailableSpace) -> bool {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/widget/text.rs:52]: Very nice!

**Review by alice-i-cecile — APPROVED** [review]
Very nice, thanks for splitting these up :)

**Review by ickshonpe — COMMENTED** [review]
Ran some naive benchmarks, saw a 12% FPS improvement over main with:
```cargo run --example many_buttons --release -- --recompute-text --no-borders --image-freq 0```

Didn't notice any significant differences without `--recompute-text` or in `many_glyphs`, as expected. 
Seems good.

**Review by ickshonpe — APPROVED** [review]
Can't find any problems.

**Pull request merged**